builder/builder-source-archive.c

```diff
@@ -234,6 +236,17 @@ get_download_location (BuilderSourceArchive *self,
       return FALSE;
     }
 
+  sources_dirs = builder_context_get_sources_dirs (context);
+  for (i = 0; i < sources_dirs->len; i++)
+    {
+      GFile* sources_root = g_ptr_array_index (sources_dirs, i);
+      g_autoptr(GFile) download_dir = g_file_get_child (sources_root, "downloads");
+      g_autoptr(GFile) sha256_dir = g_file_get_child (download_dir, self->sha256);
+      g_autoptr(GFile) file = g_file_get_child (sha256_dir, base_name);
```

Comment:
You should re-use the existing variables with the same name, instead of shadowing their definition.